TAO/TAO_IDL/driver/drv_args.cpp

```diff
@@ -688,21 +684,66 @@ process_long_option(long ac, char **av, long &i)
             {
               invalid_argument = true;
               ACE_ERROR ((LM_ERROR,
-                ACE_TEXT ("\"%C\" is not a valid argument.\n"),
+                ACE_TEXT ("\"%C\" is not a valid argument to --unknown-annotations.\n"),
                 av[i]
                 ));
             }
         }
       if (invalid_argument)
         {
           ACE_ERROR ((LM_ERROR,
-            ACE_TEXT ("Use either \"warn-once\", \"warn-all\", ")
-            ACE_TEXT ("\"error\" or \"ignore\".\n"),
+            ACE_TEXT ("Use \"warn-once\", \"warn-all\", \"error\" or \"ignore\".\n"),
             av[i]
             ));
           idl_global->parse_args_exit (1);
         }
     }
+  else if (!ACE_OS::strcmp (long_option, "preprocessor-input"))
+    {
+      bool invalid_argument = no_more_args;
+      if (no_more_args)
+        {
+          ACE_ERROR ((LM_ERROR,
+            ACE_TEXT ("--preprocessor-input is missing its required argument.")));
+        }
+      else
+        {
+          i++;
+          if (!ACE_OS::strcmp (av[i], "guess"))
+            {
+              idl_global->preprocessor_input_ = IDL_GlobalData::PreprocessorInputGuess;
+            }
+          else if (!ACE_OS::strcmp (av[i], "direct-with-e"))
+            {
+              idl_global->preprocessor_input_ = IDL_GlobalData::PreprocessorInputDirectWithE;
+            }
+          else if (!ACE_OS::strcmp (av[i], "direct-without-e"))
+            {
+              idl_global->preprocessor_input_ = IDL_GlobalData::PreprocessorInputDirectWithoutE;
+            }
+          else if (!ACE_OS::strcmp (av[i], "direct-gcc"))
+            {
+              idl_global->preprocessor_input_ = IDL_GlobalData::PreprocessorInputDirectGcc;
+            }
+          else if (!ACE_OS::strcmp (av[i], "copy"))
+            {
+              idl_global->preprocessor_input_ = IDL_GlobalData::PreprocessorInputCopy;
+            }
+          else
+            {
+              invalid_argument = true;
+              ACE_ERROR ((LM_ERROR,
+                ACE_TEXT ("\"%C\" is not a valid argument to --preprocessor-input.\n"), av[i]));
+            }
+        }
+      if (invalid_argument)
+        {
+          ACE_ERROR ((LM_ERROR, ACE_TEXT ("Use \"guess\", \"direct-with-e\", ")
+            ACE_TEXT ("\"direct-without-e\", \"direct-gcc\", or \"copy\".\n"),
+            av[i]));
```

Comment:
Is there supposed to be a `%C` or similar in the format string?

Reply:
No, it's a old typo I copied from the code for `--unknown-annotations`. I will fix both.